app/components/button.jsx

```diff
@@ -0,0 +1,231 @@
+        children
+    } = props;
+
+    const [isDisabled, setIsDisabled] = useState(disabled);
+    const [parentIsHovered, setParentIsHovered] = useState(false);
+    // Stretch Goal 
+    // const [longPress, setLongPress] = useState(false);
+    // const [position, setPosition] = useState({ x: 0, y: 0 });
+    // const longPressTimer = useRef(null);
+
+    const classes = buttonStyles();
+    const combinedClassName = `${classes[className]} ${className}`;
+
+    // Stretch Goal 
+    // const startLongPress = () => {
+    //     const x = event.clientX || (event.touches && event.touches[0].clientX);
+    //     const y = event.clientY || (event.touches && event.touches[0].clientY);
+        
+    //     setPosition({ x, y });
+    //     longPressTimer.current = setTimeout(() => {
+    //         setLongPress(true);
+    //     }, 1000);
+    // }
+
+    // const stopLongPress = (event) => {
+    //     clearTimeout(longPressTimer.current);
+    //     if (longPress) {
+    //         setLongPress(false);
+    //     }
+    // }
+
+    return (
+        <div 
+            style={{ position: 'relative' }}
+            onMouseEnter={() => setParentIsHovered(true)}
+            onMouseLeave={() => setParentIsHovered(false)}
+        >
+            <button
+                className={`${combinedClassName} ${parentIsHovered ? 'hover' : ''}`}
+                style={style}
+                title={title}
+                disabled={isDisabled}
+                onClick={() => {
+                    if (onDone) onDone();
+                    if (disableOnClick) setIsDisabled(true);
+                }}
+
+                // onMouseDown={startLongPress}
+                // onMouseUp={stopLongPress}
+                // onMouseLeave={stopLongPress}
+                // onTouchStart={startLongPress}
+                // onTouchEnd={stopLongPress}
+            >
+                {children}
+            </button>
+            {/* Stretch Goal 
+            {longPress && 
+                <div style={{
+                    top: `${position.y}px`,
+                    left: `${position.x}px`,
+                    transform: 'translateX(-50%, -100%)',
+                    backgroundColor: 'white',
+                    border: '1px solid black',
+                    padding: '5px',
+                    borderRadius: '5px',
+                }}>
+                    {title}
+                </div>
+            } */}
+        </div>
+    )
+}
+
+function ModifierButton(props) {
+    return <Button {...props} className="modifierButton" />;
+}
+
+function SecondaryButton(props) {
+    return <Button {...props} className="secondaryButton" />;
+}
+
+function PrimaryButton(props) {
+    return <Button {...props} className="primaryButton" />;
+}
+
+function TextButton(props) {
+    return <Button {...props} className="textButton" />;
+}
+
+
+const commonButtonStyles = {
+    width: 'auto',
```

Comment:
make commonButtonStyles part of buttonStyles and apply the class to the base button. 
you are going to have to pull secondaryButton, modifierButton, etc out of buttonStyles and make styles specific for each button. But it doesn't have to have ...commonButtonStyles because that will be a build in part of the Button.